src/sessionapi.psm1

```diff
@@ -1,1514 +1,1413 @@
-    [CmdletBinding()]
-    Param(
-        [parameter(Mandatory, Position = 0)]
-        [string]
-        $FilePath,
-        [parameter(Mandatory, Position = 1)]
-        [string]
-        $TargetVersion
-    )
-
-    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
-    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
-
-    if($TargetVersion -eq (Get-SafeguardSpsVersion))
-    {
-        Write-Host "$TargetVersion is already installed"
-        return
-    }
-
-    $activity = "Installing SPS upgrade"
-    Write-Progress -Activity $activity -Status 'Importing firmware' -PercentComplete 15
-    Write-Verbose "Starting firmware upload..."
-    Import-SafeguardSpsFirmware $FilePath
-    Write-Verbose "Firmware upload complete."
-    $slots = (Get-SafeguardSpsFirmwareSlot).items.body
-    for($i = 0; $i -lt $slots.count; $i++)
-    {
-        if($slots[$i].version -ieq $TargetVersion)
-        {
-            Write-Verbose "Found target firmware '$($TargetVersion)' in slot $i"
-            Write-Progress -Activity $activity -Status "Testing firmware in slot $i" -PercentComplete 65
-            if( Test-SafeguardSpsFirmware -Slot $i )
-            {
-                Write-Progress -Activity $activity -Status "Installing $TargetVersion from slot $i" -PercentComplete 75
-                Write-Verbose "Installing firmware in slot $i"
-                Install-SafeguardSpsFirmware -Slot $i -Message "Upgrading SPS firmware to $TargetVersion"
-                Write-Progress -Activity $activity -Status "Finished" -PercentComplete 100
-                Start-Sleep 60
-                Write-Verbose "Waiting for SPS to restart..."
-                for($i = 0; $i -lt 20; $i++)
-                {
-                    try 
-                    {
-                        $currentVersion = Get-SafeguardSpsVersion
-                        if($currentVersion -eq $TargetVersion)
-                        {
-                            Write-Host "Upgrade complete: SPS is at version $currentVersion"
-                            return
-                        }
-                    }
-                    catch {
-                    }
-                    Start-Sleep 15
-                }
-                throw "Timed out waiting for SPS to reach version $TargetVersion"
-            }
-            else 
-            {
-                throw "Firmware at slot $i failed upgrade test. For details run: Test-SafeguardSpsFirmware -Slot $i"
-            }
-        }
-    }
-    throw "Firmware with version $TargetVersion could not be found in any firmware slot."
-}
-
-<#
-.SYNOPSIS
-Get-SafeguardSpsSupportBundle
-
-.DESCRIPTION
-This command downloads an SPS support bundle.
-
-.PARAMETER OutFile
-The output file name. If this is omitted, a unique name will be generated.
-
-.EXAMPLE
-Get-SafeguardSpsSupportBundle  
-#>
-function Get-SafeguardSpsSupportBundle
```

Comment:
I've only just taken a quick look at this, but why has this `Get-SafeguardSpsSupportBundle` method been removed?

Reply:
It looks like I accidentally removed it from the end somehow, I added it back on the new commit.